src/validation/fields/helper/date.validation.helper.ts

```diff
@@ -43,6 +44,41 @@ export const conditionalDateValidations = (dateContextWithCondition: dateContext
   ];
 };
 
+export const conditionalHistoricalBODateValidations = (trustDateContext: dateContextWithCondition) => [
+  setValidateCeasedDateFlag(trustDateContext),
+  ...conditionalDateValidations(trustDateContext),
+];
+
+// export const setValidateCeasedDateFlag = (trustDateContext: dateContextWithCondition) => [
+//   body()
+//     .custom((value, { req }) => {
+//       const isUnableToProvideAllTrustInfo: boolean = isUnableToObtainAllTrustInfo(req);
+//       let validateCeasedDate = true;
+//       if ((req.body[trustDateContext.dayInput.name] === "" && req.body[trustDateContext.monthInput.name] === "" && req.body[trustDateContext.yearInput.name] === "") && isUnableToProvideAllTrustInfo) {
+//         validateCeasedDate = false;
+//       }
+//       req.body['validateCeasedDate'] = validateCeasedDate.toString();
+//       return true;
+//     }),
+// ];
+// export const getApplicationData = (session: Session | undefined): ApplicationData => {
+//   return session?.getExtraData(APPLICATION_DATA_KEY) || {} as ApplicationData;
+// };
+export const setValidateCeasedDateFlag = (trustDateContext: dateContextWithCondition) => {
+  console.log("Hello!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!");
+  return (req, resp, next: NextFunction) => {
+    console.log("Hello Again!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!");
+    const isUnableToProvideAllTrust: boolean = isUnableToObtainAllTrustInfo(req);
+    console.log("*****************  isUnableToObtainAllTrustInfo=" + isUnableToProvideAllTrust + " *************************");
+    let validateCeasedDate = true;
+    if ((req.body[trustDateContext.dayInput.name] === "" && req.body[trustDateContext.monthInput.name] === "" && req.body[trustDateContext.yearInput.name] === "") && isUnableToProvideAllTrust) {
+      validateCeasedDate = false;
+    }
+    console.log("*****************  validateCeasedDate=" + validateCeasedDate + " *************************");
+    req.body['validateCeasedDate'] = validateCeasedDate.toString();
```

Comment:
you could just set valildateCeasedDate to "true" or "false" then you wouldn't need to use toString()

Reply:
True. I set it to a boolean as technically it's a flag but don't mind either way.